includes/class-wc-payments-order-success-page.php

```diff
@@ -255,6 +258,70 @@ private function format_addtional_thankyou_order_received_text( string $addition
 		return sprintf( '<div class="woocommerce-info">%s</div>', $additional_text );
 	}
 
+	/**
+	 * Replace the order received text with a failure message when the order status is 'failed'.
+	 *
+	 * @param string $text The original thank you text.
+	 * @return string
+	 */
+	public function replace_order_received_text_for_failed_orders( $text ) {
+		global $wp;
+
+		$order_id = absint( $wp->query_vars['order-received'] );
+		$order    = wc_get_order( $order_id );
+
+		if ( ! $order ||
+			! $order->needs_payment() ||
+			0 !== strpos( $order->get_payment_method(), WC_Payment_Gateway_WCPay::GATEWAY_ID )
```

Comment:
I think the `REDIRECT_PAYMENT_METHODS` check should be restored, just to make sure this doesn't causes any conflicts with vouchers (e.g. multibanco).

Reply:
Nevermind, I don't think this comment is accurate, since multibanco should set the order to `on-hold`, and only `failed` and `pending` will need payment. https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-order.php#L1788

Reply:
Ah ok. Thanks for checking and clarifying!